config/diff.go

```diff
@@ -112,7 +120,8 @@ func prune(in interface{}) interface{} {
 
 		for k, v := range inType {
 			inType[k] = prune(v)
-			if inType[k] == nil {
+			_, ok := includeNilValuesForKeys[k]
```

Comment:
Same as above, I don't think we should check for specific keys in a generic function, it seems more like we need to solve this in a generic way

Reply:
ok, generic solution would be removing this block:
```
if inType[k] == nil && !ok {
				delete(inType, k)
			}
```. Then, we would output `null` for options that are set to nil. PTAL, changed it and adjusted unit tests.